app/components/SpeakingTimeContent.tsx

```diff
@@ -54,10 +54,14 @@ export default function SpeakingTimeContent({
   const formRef = useRef<HTMLFormElement>(null);
   const { t } = useTranslation("searchingCoach");
   const [isActive, setIsActive] = useState(false);
+  var timeout: NodeJS.Timeout;
 
   const handleChange = () => {
     if (formRef) {
-      submit(formRef.current, { replace: true, preventScrollReset: true });
+      // not the best solution as the tags now will timeout for 300ms as well - but with the onchange on the form I could not find another way
+      // to debounce the search input to not sent a request for each input .. happy for suggestions
+      clearTimeout(timeout)
```

Comment:
@HerrBertling Any thought about this? I did want to reuse the existing filtering through the url and implemented the searchbar the same way. But since we update on each change, the search was happening for each of the character inputs. To not introduce any new things there and also not introduce additional overhead I went with a quite pragmatic approach of debouncing the input for 300ms